### PR TITLE
Fix resource leak of file desc in test code

### PR DESCRIPTION
**Pull request by andresag01**
## Description
Fix leak of file descriptor in test code on failure (found by coverity).

## Status
**READY**

## Requires Backporting
Yes
Which branch?
2.7

### CODE REVIEW

File: tests/suites/helpers.function [hunk 1]
```diff
@@ -175,6 +175,7 @@ static int redirect_output( FILE** out_stream, const char* path )
 
     if( *out_stream == NULL )
     {
+        close( stdout_fd );
```
Comment (k-stachowiak) [inline tests/suites/helpers.function]: Could you please explain, why this `fd` should be closed here, when we have a `fclose` call earlier? It seems that the file associated with this file descriptor has already been closed, but I may be missing some nuance here.

Comment (andresag01) [reply tests/suites/helpers.function]: I think the fclose() call above closes *out_stream, but not the file descriptor in stdout_fd that is a copy of *out_stream created using dup() in line 165. Take a look at the internal reference IOTSSL-2624 for information on the issue reported by coverity.

Comment (k-stachowiak) [reply tests/suites/helpers.function]: Thanks for the answer. I can now see, that, the FD is duped, and therefore needs to be closed separately.

**Comment by sbutcher-arm** [comment]
Jenkins CI is failing on the BSD timing test, but that's as good as a pass.

Therefore promoting to 'ready for merge'.

**Pull request merged**